File: backend/src/main/java/com/tactio/mdm/domain/entity/Policy.java
package com.tactio.mdm.domain.entity;

import com.tactio.mdm.domain.enums.UpdatePolicyType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "policies")
public class Policy extends BaseEntity {

    @Column(nullable = false, unique = true, length = 120)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(nullable = false)
    private boolean passwordRequired = true;

    @Column(nullable = false)
    private int minPasswordLength = 6;

    @Column(nullable = false)
    private long maxInactivityLockMs = 60_000;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private UpdatePolicyType updatePolicy = UpdatePolicyType.WINDOWED;

    @Column(nullable = false)
    private boolean cameraDisabled = false;

    @Column(nullable = false)
    private boolean screenCaptureDisabled = false;

    @Column(nullable = false)
    private boolean factoryResetDisabled = true;

    @Column(nullable = false)
    private boolean installAppsDisabled = false;

    @Column(nullable = false)
    private boolean usbFileTransferDisabled = false;

    @Column(columnDefinition = "TEXT")
    private String restrictionsJson;

    @Column(nullable = false)
    private boolean active = true;
}

File: frontend/src/types/policy.ts
import type { PolicyTargetType, UpdatePolicyType } from "./enums";

export interface PolicyResponse {
  id: string;
  name: string;
  description: string | null;
  passwordRequired: boolean;
  minPasswordLength: number;
  maxInactivityLockMs: number;
  updatePolicy: UpdatePolicyType;
  cameraDisabled: boolean;
  screenCaptureDisabled: boolean;
  factoryResetDisabled: boolean;
  installAppsDisabled: boolean;
  usbFileTransferDisabled: boolean;
  restrictionsJson: string | null;
  active: boolean;
}

export interface PolicyRequest {
  name: string;
  description?: string;
  passwordRequired: boolean;
  minPasswordLength: number;
  maxInactivityLockMs: number;
  updatePolicy: UpdatePolicyType;
  cameraDisabled: boolean;
  screenCaptureDisabled: boolean;
  factoryResetDisabled: boolean;
  installAppsDisabled: boolean;
  usbFileTransferDisabled: boolean;
  restrictionsJson?: string;
  active: boolean;
}

export interface PolicyAssignmentResponse {
  id: string;
  policyId: string;
  policyName: string;
  targetType: PolicyTargetType;
  deviceId: string | null;
  departmentId: string | null;
  tagId: string | null;
}

export interface PolicyAssignmentRequest {
  policyId: string;
  targetType: PolicyTargetType;
  targetId: string;
}
